### PR TITLE
changed name of neighbors argument

### PR DESCRIPTION
**Pull request by melodyyzh**
## Description
Changed `neighbors` to `env_neighbors` in `EnvironmentMotifMatch`. 
Changed `env_neighbors` to `cluster_neighbors` in `EnvironmentCluster`. 

## Motivation and Context
Resolves: #989

## How Has This Been Tested?
We will update tests for the right keyword arguments 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds or improves functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation improvement (updates to user guides, docstrings, or developer docs)

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/glotzerlab/freud/blob/master/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [ ] I have updated the documentation (if relevant).
- [x] I have added tests that cover my changes (if relevant).
- [x] All new and existing tests passed.
- [ ] I have updated the [credits](https://github.com/glotzerlab/freud/blob/master/doc/source/reference/credits.rst).
- [x] I have updated the [Changelog](https://github.com/glotzerlab/freud/blob/master/ChangeLog.md).

### CODE REVIEW

**Review by tommy-waltmann — CHANGES_REQUESTED** [review]
Great work so far on this @melodyyzh!

There are a few things still to fix before this PR is ready for merge:

1) There are still a few instances in the docstring for the `EnvironmentCluster` class's `compute` method where the word `neighbors` needs to be changed to `cluster_neighbors`.

2) On #989, Vyas had some comments about improving the parameter descriptions and reworking the internal documentation of the C++ code. I can help with this if needed.

3) Part of contributing to an open-source project like freud is that you get to give yourself some credit. In the file `doc/source/reference/credits.rst`, you can add an entry for your name and say what you contributed to the project.

As always, let us know if you need any help.

File: freud/environment.pyx [hunk 1]
```diff
@@ -664,7 +664,7 @@ cdef class EnvironmentCluster(_MatchEnv):
             warnings.warn(
                 "The global search option is deprecated and will be removed in "
                 "version 3.0. If you want this behavior, use a NeighborList "
-                "composed of all pairs of particles in the system as the `neighbors`.",
+                "composed of all pairs of particles in the system as the `cluster_neighbors`.",
                 FutureWarning
             )
```
Comment (melodyyzh) [inline freud/environment.pyx]: I remembered to test the branch after I pushed the changes. All the tests passed. I saw that there is a conflict waiting to be solved. I am not sure if that is caused by this?

Comment (tommy-waltmann) [reply freud/environment.pyx]: Don't worry about merge conflicts right now, I can fix those. It says there are conflicts in two flies, so its not just the line above.

File: doc/source/reference/credits.rst [hunk 2]
```diff
@@ -376,6 +376,9 @@ Alain Kadar
 
 * Introduced mass dependence for ``ClusterProperties`` class, inertia tensor and radius of gyration.
 
+Melody Zhang
+
+* Changed ``neighbors`` argument to ``env_neighbors`` for ``EnvironmentMotifMatch`` class and to ``cluster_neighbors`` for ``EnvironmentCluster`` class.
 Source code
```
Comment (tommy-waltmann) [inline doc/source/reference/credits.rst]:
```suggestion
* Changed ``neighbors`` argument to ``env_neighbors`` for ``EnvironmentMotifMatch`` class and to ``cluster_neighbors`` for ``EnvironmentCluster`` class.

Source code
```
Let's keep a space in between sections here, as that was the formatting before.

**Review by tommy-waltmann — CHANGES_REQUESTED** [review]
There are two things left to do on this PR:

1) Update the Changelog to mention the new argument name.

2) Update the migration guide with a small example showing the new argument name.

I can assist with either of these tasks, @melodyyzh . Great work so far, this is almost ready for merge.

**Review by tommy-waltmann — CHANGES_REQUESTED** [review]
@melodyyzh this PR is basically ready for merge, except there is one check that is failing right now. Pre-commit has identified a small change to the code format in one file that needs to be addressed. It should be a really quick change to make and then your first ever freud PR can be merged!

**Comment by joaander** [comment]
> @melodyyzh this PR is basically ready for merge, except there is one check that is failing right now. Pre-commit has identified a small change to the code format in one file that needs to be addressed. It should be a really quick change to make and then your first ever freud PR can be merged!

@melodyyzh to clarify, you don't need to make this fix manually. You can ask the pre-commit bot to do it (see instructions at [the bottom of the status page](https://results.pre-commit.ci/run/github/167217991/1674842126.__nfB7klSG-BVlpG4WaZTA)) or you can install and execute `pre-commit run --all-files` in your local clone, then commit and push the changes (see the [pre-commit docs for more details](https://pre-commit.com/)).

**Comment by melodyyzh** [comment]
pre-commit.ci autofix

**Review by tommy-waltmann — APPROVED** [review]
Looks good, thanks for all your work on this!

**Pull request merged**